src/main/java/com/schedule/workout/workoutScheduler/controller/workout/WorkoutsController.java

```diff
@@ -0,0 +1,98 @@
+import com.schedule.workout.workoutScheduler.exceptions.WorkoutNotFoundException;
+import com.schedule.workout.workoutScheduler.services.WorkoutsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+
+@Controller
+public class WorkoutsController {
+    @Autowired
+    private WorkoutsService workoutsService;
+
+    @RequestMapping(method = RequestMethod.POST, value = "/workouts")
+    public ResponseEntity<CreateWorkoutModel> createWorkout(@Valid @RequestBody CreateWorkoutModel createWorkoutModel){
+        try {
+            return ResponseEntity
+                    .status(HttpStatus.CREATED)
+                    .body(workoutsService.createWorkout(createWorkoutModel));
+        } catch (InvalidWorkoutBodyException e){
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .build();
+        }catch (UserNotFoundException e){
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .build();
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/workouts")
+    public ResponseEntity<List<WorkoutModel>> getAllWorkouts(@RequestParam(name = "name",required = false)String name,
+                                                             @RequestParam(name = "description",required = false)String description,
+                                                             @RequestParam(name = "duration",required = false) Integer duration,
+                                                             @RequestParam(name = "difficulty",required = false)Integer difficulty
+                                                            ){
+        try {
+            return ResponseEntity
+                    .status(HttpStatus.OK)
+                    .body(workoutsService.getAllWorkouts(name, description, duration, difficulty));
+        }catch (WorkoutNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .build();
+            }
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/workouts/{id}")
+    public ResponseEntity<WorkoutModel> getWorkoutById(@PathVariable String id) {
+        try {
+            return ResponseEntity
+                    .status(HttpStatus.OK)
+                    .body(workoutsService.getWorkoutById(id));
+        }catch (WorkoutNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .build();
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.PUT, value = "/workouts/{id}")
+    public ResponseEntity<UpdateWorkoutModel> updateWorkout(@PathVariable String id, @Valid @RequestBody UpdateWorkoutModel updateWorkoutModel) {
```

Comment:
If you return NO_CONTENT, then you should return ResponseBody. ResponseBody<> is only used when you want to return json body